### PR TITLE
Clean up Helm chart documentation

### PR DESCRIPTION
**Pull request by plumbis**
### Description of your changes
This PR cleans up and expands the documentation for the Crossplane Helm chart. 

This will simplify the process of copying the Helm chart values to the [docs](https://docs.crossplane.io/master/software/install/#customize-the-crossplane-helm-chart). 

Relates to [crossplane/docs #401](https://github.com/crossplane/docs/issues/401)

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

Updated content passes docs Vale linting. 

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by hasheddan — COMMENTED** [review]
I believe that @phisco implemented support to generate this in https://github.com/upbound/build/pull/230 -- would that work for what you are looking for @plumbis?

File: cluster/charts/crossplane/values.yaml [hunk 1]
```diff
@@ -1,213 +1,218 @@
...
 
 registryCaBundleConfig:
-  # -- Name of ConfigMap containing additional CA bundle for fetching from package registries.
-  name: {}
-  # -- Key to use from ConfigMap containing additional CA bundle for fetching from package registries.
-  key: {}
+  # -- The ConfigMap name containing a custom CA bundle to enable fetching packages from registries with unknown or untrusted certificates.
+  name: ""
+  # -- The ConfigMap key containing a custom CA bundle to enable fetching packages from registries with unknown or untrusted certificates.
+  key: ""
```
Comment (phisco) [inline cluster/charts/crossplane/values.yaml:55]: Reminder to myself: I have to check how these were used and that there wasn't a specific reason for this to be a `{}`, happy to have it have the right type.

File: cluster/charts/crossplane/values.yaml [hunk 2]
```diff
@@ -1,213 +1,218 @@
...
-  # -- Opt out of deploying aggregated ClusterRoles.
+  # -- Don't install aggregated Crossplane ClusterRoles.
   skipAggregatedClusterRoles: false
-  # -- The number of replicas to run for the RBAC Manager pods.
+  # -- The number of RBAC Manager pod `replicas` to deploy.
   replicas: 1
-  # -- The extent to which the RBAC manager will manage permissions:.
-  # - `All` indicates to manage all Crossplane controller and user roles.
-  # - `Basic` indicates to only manage Crossplane controller roles and the `crossplane-admin`, `crossplane-edit`, and `crossplane-view` user roles.
+  # -- Defines the roles the RBAC Manager creates Roles and ClusterRoles
```
Comment (phisco) [inline cluster/charts/crossplane/values.yaml:68]: Not exactly sure what you wanted to say, maybe:
```suggestion
  # -- Defines the extent to which the RBAC Manager creates Roles and ClusterRoles
```

File: cluster/charts/crossplane/values.yaml [hunk 3]
```diff
@@ -1,213 +1,218 @@
...
 
 packageCache:
-  # -- Storage medium for package cache. `Memory` means volume will be backed by tmpfs, which can be useful for development.
+  # -- Set to `Memory` to hold the package cache in a RAM-backed file system. Useful for Crossplane development.
   medium: ""
-  # -- Size limit for package cache. If medium is `Memory` then maximum usage would be the minimum of this value the sum of all memory limits on containers in the Crossplane pod.
+  # -- The size limit for the package cache. If medium is `Memory` the `sizeLimit` can't exceed Node memory.
   sizeLimit: 20Mi
-  # -- Name of the PersistentVolumeClaim to be used as the package cache. Providing a value will cause the default emptyDir volume to not be mounted.
+  # -- The name of a PersistentVolumeClaim to use as the package cache. Disables the default package cache `emptyDir`.
```
Comment (phisco) [inline cluster/charts/crossplane/values.yaml:113]: I found it a bit confusing, maybe:
```suggestion
  # -- The name of a PersistentVolumeClaim to use as the package cache. Setting it to `emptyDir` disables the default package cache.
```

File: cluster/charts/crossplane/values.yaml [hunk 4]
```diff
@@ -1,213 +1,218 @@
...
   pvc: ""
-  # -- Name of the ConfigMap to be used as package cache. Providing a value will cause the default emptyDir volume not to be mounted.
+  # -- The name of a ConfigMap to use as the package cache. Disables the default package cache `emptyDir`.
```
Comment (phisco) [inline cluster/charts/crossplane/values.yaml:115]: I found it a bit confusing, maybe:
```suggestion
  # -- The name of a ConfigMap to use as the package cache. Setting it to `emptyDir` disables the default package cache.
```

File: cluster/charts/crossplane/values.yaml (hunk 1 above)
Comment (phisco) [reply cluster/charts/crossplane/values.yaml:55]: All good 🎉

File: cluster/charts/crossplane/values.yaml (hunk 2 above)
Comment (plumbis) [reply cluster/charts/crossplane/values.yaml]: Pushed a change, do you think this is better?

File: cluster/charts/crossplane/values.yaml (hunk 3 above)
Comment (phisco) [reply cluster/charts/crossplane/values.yaml:113]: I totally misunderstood it, sorry, but yes, adding `Volume` would make it clearer 👍

Comment (plumbis) [reply cluster/charts/crossplane/values.yaml:113]: Fixed!

File: cluster/charts/crossplane/values.yaml (hunk 2 above)
Comment (phisco) [reply cluster/charts/crossplane/values.yaml]: I'm not an expert on the RBAC Manager management policies, I had a look at the code and the design doc, but I'd rather have someone with more context to judge https://github.com/crossplane/crossplane/pull/4085/files#diff-fbc48c3c51c7fab207110c647178d455b5677a3a0042592b005e849d86c46f1cR68 , maybe you can have a look at it @hasheddan?

File: cluster/charts/crossplane/values.yaml [hunk 5]
```diff
@@ -1,213 +1,218 @@
...
   skipAggregatedClusterRoles: false
-  # -- The number of replicas to run for the RBAC Manager pods.
+  # -- The number of RBAC Manager pod `replicas` to deploy.
   replicas: 1
-  # -- The extent to which the RBAC manager will manage permissions:.
-  # - `All` indicates to manage all Crossplane controller and user roles.
-  # - `Basic` indicates to only manage Crossplane controller roles and the `crossplane-admin`, `crossplane-edit`, and `crossplane-view` user roles.
+  # -- Defines the Roles and ClusterRoles the RBAC Manager creates and manages.
+  # - A policy of `Basic` creates and binds Roles only for the Crossplane ServiceAccount, Provider ServiceAccounts and creates Crossplane ClusterRoles.
+  # - A policy of `All` includes all the `Basic` settings and also creates Crossplane Roles in all namespaces.
```
Comment (phisco) [inline cluster/charts/crossplane/values.yaml:70]: My only doubt is whether we should explicitly mention the roles names as we did before here.

**Review by phisco — APPROVED** [review]
LGTM overall, minor doubt on the RBAC manager `managementPolicy`, but I'll leave that to the next reviewer to decide.

File: cluster/charts/crossplane/values.yaml (hunk 5 above)
Comment (phisco) [reply cluster/charts/crossplane/values.yaml:70]: To make it clearer, I meant mentioning explicitly`crossplane-admin`, `crossplane-edit`, and `crossplane-view` as we were doing before.

**Review by negz — APPROVED** [review]
Few thoughts but overall looks good.

**Comment by plumbis** [comment]
I've borked my branch, closing and opening another PR.

**Pull request closed without merging**